server/src/main/java/com/linecorp/centraldogma/server/internal/api/DtoConverter.java

```diff
@@ -58,29 +60,38 @@ public static RepositoryDto convert(Repository repository) {
                                  repository.creationTimeMillis());
     }
 
-    public static <T> EntryDto<T> convert(Repository repository, Revision revision,
-                                          Entry<T> entry, boolean withContent) {
+    public static EntryDto convert(Repository repository, Revision revision,
+                                   Entry<?> entry, boolean withContent) {
+        return convert(repository, revision, entry, withContent, null);
+    }
+
+    public static EntryDto convert(Repository repository, Revision revision,
+                                   Entry<?> entry, boolean withContent, @Nullable QueryType queryType) {
         requireNonNull(entry, "entry");
         if (withContent && entry.hasContent()) {
-            return convert(repository, revision, entry.path(), entry.type(), entry.content());
+            // TODO(ks-yim): Not sure of 'DtoConverter' having business logic and it looks hacky to receive
+            //   queryType to handle JSON_PATH query for JSON5 files.
+            return convert(repository, revision, entry.path(), entry.type(),
+                           isJson5(entry) && queryType != QueryType.JSON_PATH ? entry.contentAsText()
```

Comment:
Q) Did we agree to move this functionality to api v2?
Since `type == JSON` and `content` is `json5`, I feel like this will break backwards-compatibility

Reply:
Haven't talked about V2 part, yet.

> Since type == JSON and content is json5, I feel like this will break backwards-compatibility

I don't think there will be a breaking change with `ArmeriaCentralDogma` client unless users are using deprecated `Query.identity(...)` because JSON5 can now only be queried via `Query.ofText(..)` and this PR still supports it.

The only breaking change that I can think of is that users won't be able to upsert JSON5 text with `Change.ofTextUpsert(...)` after this PR on anymore, though I doubt there will be many users who do this.

But I am not sure if this PR won't break admin UI and its APIs somehow.
Let me check this out.

Reply:
> I was thinking of the following scenario:

Oh, I've thought CD client's `getFile` API always accept `Query` but it seems not.
There could be a compatibility issue if a user uses the following API.

The return type of `Entry#content()` retrieved via this API is no longer `String` but `TextNode` for JSON5 files.
```java
    default CompletableFuture<Entry<?>> getFile(String projectName, String repositoryName,
                                                Revision revision, String path) {
        @SuppressWarnings("unchecked")
        final CompletableFuture<Entry<?>> f = (CompletableFuture<Entry<?>>) (CompletableFuture<?>)
                getFile(projectName, repositoryName, revision, Query.of(QueryType.IDENTITY, path));
        return f;
    }
```

Reply:
Also had a chat about this and decided to go for `ContentServiceV2` which will be handled in another PR.